File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/request/StoreSearchRequest.java
package com.coffee.gu.api.controller.v1.request;

import com.coffee.gu.CoreException;
import com.coffee.gu.ErrorType;
import com.coffee.gu.store.StoreSearch;


public record StoreSearchRequest(
        Double latitude,
        Double longitude,
        Double radiusKm
) {
    public StoreSearch toStoreSearch() {
        if (latitude == null || longitude == null || radiusKm == null) {
            throw new CoreException(ErrorType.INVALID_REQUEST, null);
        }
        return new StoreSearch(latitude, longitude, radiusKm);
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/request/StoreSearchRequest.kt
package com.coffee.gu.api.controller.v1.request

import com.coffee.gu.store.StoreSearch

class StoreSearchRequest(
    val latitude: Double,
    val longitude: Double,
    val radiusKm: Double,
) {
    fun toStoreSearch(): StoreSearch {
        return StoreSearch(latitude, longitude, radiusKm)
    }
}
